### PR TITLE
Increment minor schema version API

### PR DESCRIPTION
**Pull request by christophermlawson**

### CODE REVIEW

File: core/ecschema-editing/src/Editing/Editor.ts [hunk 1]
```diff
@@ -166,6 +166,18 @@ export class SchemaContextEditor {
     return {};
   }
 
+  /**
+   * Increments the minor version of a schema.
+   * @param schemaKey The SchemaKey identifying the schema.
+   */
+  public async incrementMinorVersion(schemaKey: SchemaKey): Promise<SchemaEditResults> {
+    const schema = (await this.schemaContext.getCachedSchema(schemaKey, SchemaMatchType.Latest)) as MutableSchema;
+    if (schema === undefined) return { errorMessage: `Schema Key ${schemaKey.toString(true)} not found in context` };
```
Comment (pmconne) [inline core/ecschema-editing/src/Editing/Editor.ts:175]: `getCachedSchema<T>` does not do any checking to confirm the object is actually an instance of `T`. Shouldn't it?

Comment (pmconne) [reply core/ecschema-editing/src/Editing/Editor.ts:175]: The schema editing package appears to be littered with this kind of code - e.g., every caller of `Schema.lookupItem`.
You are aware that `as` [does not perform a dynamic cast](https://blogs.bentley.com/2019/10/17/typescript-gotchas-dynamic-casts/), right?

Comment (christophermlawson) [reply core/ecschema-editing/src/Editing/Editor.ts:175]: Regarding you first comment, it comes down to our hackish approach of all the 'Mutable' abstract classes that we use to access the protected methods of a given class.  The class Schema has a corresponding abstract class MutableSchema which extends the schema class.  As long as it's a Schema instance, which is must be, we can always cast it to a MutableSchema to access the protected methods (again hackish).  Colin and I spoke about the Mutable classes the other day and we are thinking about getting rid of them.  They were originally written, for internal use, to handle Schema deserialization and testing purposes.  Relying on them for schema editing as well, may be a mistake.  I will take a look at the other cases where we are using the 'as' operator.  Thanks for the feedback.

Comment (christophermlawson) [reply core/ecschema-editing/src/Editing/Editor.ts:175]: I've looked at other cases of dynamic casting, and I agree that in some of the cases we are open to accepting incorrect types leading to unexpected errors.  I've created a backlog item to look at each case.  In most cases we know what we have (like by checking the SchemaItemType of the object) before casting which just allows us to access the type's member prop/method.

**Review by ColinKerr — APPROVED** [review]

File: common/api/ecschema-metadata.api.md [hunk 2]
```diff
@@ -483,7 +503,11 @@ export const Diagnostics: {
             readonly diagnosticType: DiagnosticType;
             ecDefinition: AnyClass;
             messageArgs?: [string, string] | undefined;
-            category: DiagnosticCategory;
+            category: DiagnosticCategory; /**
+             * The list of [[IDiagnostic]] implementation classes used by the EC rule implementations.
```
Comment (calebmshafer) [inline common/api/ecschema-metadata.api.md:507]: not sure the extract-api is going to like this change....

**Review by calebmshafer — CHANGES_REQUESTED** [review]

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**